bpf/aya-bpf/src/args.rs

```diff
@@ -111,6 +122,15 @@ impl<T> FromPtRegs for *const T {
         }
     }
 
+    fn from_stack_argument(ctx: &pt_regs, n: usize) -> Option<Self> {
+        unsafe {
+            let addr: __u64 = ctx.rsp + 8 * (n + 1) as __u64;
+            bpf_probe_read(addr as *const T)
+                .map(|v| &v as *const _)
+                .ok()
```

Comment:
Instead of .ok() can we just return the Result?

Reply:
I just follow `from_argument`. It seems that, `.ok()` is more readable.

Reply:
It's discarding information.

Reply:
> It's discarding information.

I'm confused. I know `bpf_probe_read` returns `Result`, and if `from_stack_argument` returns `Option`, `Err` info may be lost. 
But the exist `from_argument` just return `Option`. Should it be better to return `Result` to save information or return `Option` to adapt the style ?

Reply:
I'm not sure why the existing functions return `Option`. In my opinion they should also return `Result`. Perhaps @alessandrod knows something about the return value that I don't.

Reply:
Yeah, bpf_probe_read returns only 0 or -EFAULT, so it's not really discarding anything. 

But I'll say that this `from_argument` stuff is very involved - in the existing code I mean, not just in the PR, and needs to be reworked. Sadly I forgot how I'd rework it it's been a while :P

Reply:
if `bpf_probe_read` only returns one of two values, should we wrap it and do the Result -> Option conversion there instead of everywhere it's used?

Reply:
Maybe, but I'm inclined to say no.

My idea is we want to offer different levels of abstractions:

- Generated bindings that give you the original (unsafe) helpers API. This is useful when you want to call things for which we don't provide safe or higher level APIs yet.
- Safe wrappers for the generated helpers. These are still low level, but safe. We don't control these APIs, we just wrap them, and they're unstable so they might change. `bpf_probe_read` might start returning something else at some point (unlikely for the bpf_probe_read case specifically but possible for a more complex helper). Additionally, say I want to write an ebpf program without using any high level bits of `aya-ebpf`, I should be able to do so, and I should be able to build my abstractions on top of these foundational helpers. 
- We have our own API. `ProbeContext::arg` is an example. It internally uses bpf_probe_read, but that's an implementation detail. If bpf_probe_read starts returning something else, it's conceivable that we won't care at this level and we'll still return None; or if we do decide to return the inner error, we can deprecated the API and have `::argument` or something, but we won't be as weird as having to deprecate `bpf_probe_read` which technically we don't control we're just wrapping.

Reply:
Yeah, I wasn't suggesting changing the behavior of `bpf_probe_read`, but as it stands we have a *lot* of `.ok()` calls that I think we could centralize.